### ISSUE TITLE
Add TSkLabel.TextSize property

### ISSUE THREAD

**Issue by havrlisan** [skia4delphi/skia4delphi#250]
TSkLabel should have a public property for retrieving text size. Currently, a private method GetFitSize(var AWidth, AHeight) is used for calculating the control size depending on its alignment. Still, there should be another one that will always return the size of the text and is independent of the control's alignment.

**Comment by checkdigits** [comment]
When you say "size of the text" do you mean "dimensions of the text"?

This is not as easy as it sounds.

For example, in this image, you can return a Rect which bounds the widest extent of the text:

![image](https://github.com/skia4delphi/skia4delphi/assets/2126394/060739fc-6759-47c8-9dd9-e203b1b97439)

which would be this kind of thing:

![image](https://github.com/skia4delphi/skia4delphi/assets/2126394/72c20a6c-f797-403b-b512-ab954b96ec41)


But what about if the text is rotated?

![image](https://github.com/skia4delphi/skia4delphi/assets/2126394/adfe2d5f-b332-452d-9373-cf79136c0d1a)

Now the Rect is like this:

![image](https://github.com/skia4delphi/skia4delphi/assets/2126394/156d8db2-c08e-4035-8e64-6868c9defcf6)

...but do you actually want this?

![image](https://github.com/skia4delphi/skia4delphi/assets/2126394/5856d036-7a09-44a9-a4bb-736cbfc16e25)

**Comment by checkdigits** [comment]
Also note that in the recent version of Skia4Delphi the labels can now mix the font sizes and styles. :)

**Comment by havrlisan** [comment]
Thanks for the fast reply, Ian. I understand your point, but what I want to achieve is simply the same size of the control when the AutoSize property is set to true. Consider this scenario:

I have a TLayout with TAlignLayout.Top, and inside the TLayout are two TSkLabel controls whose Align is set to Left (meaning AutoSize is applied to Width only). I want the TLayout.Height to be set according to its content height, which, in this case, is the bigger height of the two TSkLabel controls. However, I cannot retrieve the true text height of either of the two labels because currently there is no way to directly get it. What I can do, however, is:
- set the label's Align to None,
- retrieve the updated height (which TSkLabel retrieved from a private method GetFitSize),
- re-set the label's Align to Left.

I am aware that TSkLabel has a collection of strings (or words) instead of a single string, but what I'm asking for here is to retrieve the **total** size of the TSkLabel. Having the ability to retrieve the size of every word would be a great feature, but I cannot see it being used in almost any scenario (as pretty much anything you would want to do with a single word is available through Skia's text and decoration settings).

The same goes for text rotation. I am only asking for the AutoSize's calculated size to be available through a function/property even without AutoSize set to true.

What you displayed in the pictures above is

**Comment by havrlisan** [comment]
> Sounds like this function might not currently exist. There might be a combination of skCanvas properties which can be used but I'm not sure.
> 
> This might be one for @paulocesarbot or @viniciusfbb to answer, I think.

I've used Skia for some time now, and my request is straightforward. _This function_ can be found right [here](https://github.com/skia4delphi/skia4delphi/blob/107900bb3ca7791d7658116ecb45157581b9dcd6/Source/FMX/FMX.Skia.pas#L1080C14-L1080C14), in the _FMX.Skia.pas_ unit.

On a non-related matter, I sincerely hope that Skia4Delphi will remain 100% open-source after integrating into FMX.

**Comment by checkdigits** [comment]
> On a non-related matter, I sincerely hope that Skia4Delphi will remain 100% open-source after integrating into FMX

Yes that is correct. Skia4Delphi and the FMX integration are related but not quite the same and the Skia4Delphi project will continue to be as it is now.

The integration, of course, is not open source as such, but the source is available as it always is with the RTL and related units.
